File: Demo.CS/Examples/SerializeObject.cs
using CsvSerializer;
using Demo.CS.Models;
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;
using System.Threading.Tasks;

namespace Demo.CS.Examples
{
    public static class SerializeObject
    {
        public async static Task Serialize()
        {
            // create some data
            List<User> users = new List<User>()
            {
                new User { Username = "super user", Email = "superuser@example.com", Age = 54, DateOfBirth = DateTime.UtcNow.AddYears(-50) },
                new User { Username = "user", Email = "user@example.com", Age = 20, DateOfBirth = DateTime.UtcNow.AddYears(-20) }
            };

            // without object mapper
            using (MemoryStream memoryStream = new MemoryStream())
            {
                Serializer serializer = new Serializer();
                await serializer.SerializeObjects(users, memoryStream).ConfigureAwait(true);

                Console.WriteLine(Encoding.UTF8.GetString(memoryStream.ToArray()));
                Console.ReadLine();
            }

            // with object mapper
            using (MemoryStream memoryStream = new MemoryStream())
            {
                var mapper = new ObjectMapper<User>();
                mapper.AddMap(u => u.Username, 1, "Name")
                    .AddMap(u => u.Age, 0)
                    .AddMap(u => u.Email)
                    .AddMap(u => u.DateOfBirth, header: "Date of birth", transformFunction: dob => dob.ToString("yyyy-mm-dd"));

                Serializer serializer = new Serializer();
                await serializer.SerializeObjects(users, memoryStream, mapper).ConfigureAwait(true);

                Console.WriteLine(Encoding.UTF8.GetString(memoryStream.ToArray()));
                Console.ReadLine();
            }
        }
    }
}

File: Demo.VB/Examples/SerializeObject.vb
Imports System.IO
Imports System.Text
Imports CSVSerializer

Public Class SerializeObject
    Public Shared Async Function Serialize() As Task

        ' create some data
        Dim users = New List(of User) From {
            New User With { .Username = "super user", .Email = "superuser@example.com", .Age = 54 },
            New User With { .Username = "user", .Email = "user@example.com", .Age = 20 }
        }

        ' without object mapper
        Using memoryStream As New MemoryStream()
            Dim serializer = New Serializer()
            Await serializer.SerializeObjects(users, memoryStream).ConfigureAwait(true)

            Console.WriteLine(Encoding.UTF8.GetString(memoryStream.ToArray()))
            Console.ReadLine()
        End Using

        ' with object mapper
        Using memoryStream As New MemoryStream()
            Dim mapper = New ObjectMapper(Of User)
            With mapper
                .AddMap(Function(u) u.Username, 1, "Name")
                .AddMap(Function(u) u.Age, 0)
                .AddMap(Function(u) u.Email)
            End With

            Dim serializer = New Serializer()
            Await serializer.SerializeObjects(users, memoryStream, mapper).ConfigureAwait(true)

            Console.WriteLine(Encoding.UTF8.GetString(memoryStream.ToArray()))
            Console.ReadLine()
        End Using
    End Function
End Class
